src/auth/mas.py

```diff
@@ -2,32 +2,227 @@
+    """Exception raised when generation of admin token failed."""
+
+
+def validate_mas_config(container: ops.model.Container) -> None:
+    """Validate current MAS configuration.
+
+    Args:
+        container: Synapse container.
+    """
+    command = [MAS_EXECUTABLE_PATH, "config", "check", "-c", MAS_CONFIGURATION_PATH]
+    process = container.exec(command=command, working_dir=MAS_WORKING_DIR)
+    process.wait_output()
+
+
+def sync_mas_config(container: ops.model.Container) -> None:
+    """Sync the MAS configuration with the database.
+
+    Args:
+        container: Synapse container.
+    """
+    command = [MAS_EXECUTABLE_PATH, "config", "sync", "--prune", "-c", MAS_CONFIGURATION_PATH]
+    process = container.exec(command=command, working_dir=MAS_WORKING_DIR)
+    process.wait()
+
+
+def register_user(
+    container: ops.model.Container,
+    username: str,
+    is_admin: bool = False,
+) -> str:
+    """Register a new user with MAS.
+
+    Args:
+        container: Synapse container.
+        username: The username.
+        is_admin: Whether the user is an admin. Defaults to False.
+
+    Raises:
+        MASRegisterUserFailedError: when user registration fails
+
+    Returns:
+        str: The generated user password
+    """
+    password = secrets.token_hex(16)
+    command = [
+        MAS_EXECUTABLE_PATH,
+        "-c",
+        MAS_CONFIGURATION_PATH,
+        "manage",
+        "register-user",
+        "--yes",
+        username,
+        "--password",
+        f"'{password}'",
+    ]
+    if is_admin:
+        command.append("--admin")
+    try:
+        process = container.exec(command=command, working_dir=MAS_WORKING_DIR)
+        process.wait_output()
+    except ops.pebble.ExecError as exc:
+        logger.error("Error registering new user: %s", exc.stderr)
+        raise MASRegisterUserFailedError("Error validating MAS configuration.") from exc
 
+    return password
 
-# pylint: disable=too-few-public-methods
+
+def verify_user_email(
+    container: ops.model.Container,
+    username: str,
+    email: str,
+) -> None:
+    """Verify a user email with mas-cli.
+
+    Args:
+        container: Synapse container.
+        username: The username.
+        email: The user's email.
+
+    Raises:
+        MASVerifyUserEmailFailedError: when user registration fails
+    """
+    command = [
+        MAS_EXECUTABLE_PATH,
+        "-c",
+        MAS_CONFIGURATION_PATH,
+        "manage",
+        "verify-email",
+        username,
+        email,
+    ]
+
+    try:
+        process = container.exec(command=command, working_dir=MAS_WORKING_DIR)
+        process.wait_output()
+    except ops.pebble.ExecError as exc:
+        logger.error("Error verifying the user email: %s", exc.stderr)
+        raise MASVerifyUserEmailFailedError("Error verifying the user email.") from exc
+
+
+def generate_admin_access_token(container: ops.model.Container, username: str) -> str:
+    """Verify a user email with mas-cli.
```

Comment:
Minor: is this description correct?

Reply:
Actually I'll remove this since it's not currently being used ( it was added to support mjolnir but we removed mjolnir )